pylint/checkers/base.py

```diff
@@ -2039,13 +2043,9 @@ def _name_disallowed_by_regex(self, name: str) -> bool:
             pattern.match(name) for pattern in self._bad_names_rgxs_compiled
         )
 
+    # pylint: disable-next=arguments-renamed
```

Comment:
If we make `confidence` a keyword only parameter on `_check_name` can't we avoid this warning and allow us to add typing to it and the parameter of the other implementation of `_check_name`?

Reply:
in `non_ascii_checker` you or @Pierre-Sassoulas suggested to remove the confidence parameter and hard code it. Also I needed to introduce there another parameter optional `check_string`, so I can check the names of modules, without getting problems with the "." within the name.

If we only check for `.isascii` it actually would simplify everyhting a lot, because this would not be needed anymore.

Reply:
What about changing the signature of `check_name` to:
```python
self, node_type: str, name: str, node: nodes.NodeNG, confidence: Confidence = interfaces.UNKNOWN, check_string: Optional[str] = None,
```
Wouldn't that avoid these warnings?

Reply:
Yep it would. Worte that a different point already.

Reply:
Just did it anyway and didn't wait for @Pierre-Sassoulas, because I just want to finish it ASAP.